src/procodile/cli.cr

```diff
@@ -159,14 +159,14 @@ module Procodile
       end
     end
 
-    private def self.options(name : Symbol, &block : Proc(OptionParser, Procodile::CLI, Nil))
+    private def self.options(name : Symbol, &block : Proc(OptionParser, Procodile::CLI, Nil)) : Proc(OptionParser, Procodile::CLI, Nil)
       @@options[name] = block
     end
 
     struct Command
       getter name : String, description : String?, options : Proc(OptionParser, Procodile::CLI, Nil)?, callable : Proc(Nil)
 
-      def initialize(@name, @description, @options, @callable)
+      def initialize(@name : String, @description : String, @options : Proc(OptionParser, Procodile::CLI, Nil), @callable : Proc(Nil) | Proc(NoReturn))
```

Comment:
Current compilation error is here:

```
In src/procodile/cli.cr:169:7

 169 | def initialize(@name : String, @description : String, @options : Proc(OptionParser, Procodile::CLI, Nil), @callable : Proc(Nil) | Proc(NoReturn))
       ^
Error: expected argument 'callable' to 'Procodile::CLI::Command#initialize' to be (Proc(Nil) | Proc(NoReturn)), not (Proc(Nil) | Proc(NoReturn))

Overloads are:
 - Procodile::CLI::Command#initialize(name : String, description : String, options : Proc(OptionParser, Procodile::CLI, Nil), callable : Proc(Nil) | Proc(NoReturn))
```

:confused:

Reply:
Oh, derp, that's weird. The type restriction above on line 167 has it as `Proc(Nil)`, but for some reason `Proc(NoReturn)` comes up.... Where's that `NoReturn` coming from? And why is the compiler not very helpful in this case?

Reply:
Manually removing the `Proc(NoReturn)` lets things build and all tests pass again with this PR. Interesting.